### PR TITLE
#165305266 Implemented Create Article CRUD

### PR DESCRIPTION
**Pull request by ElishaMisoi**
### What does this PR do?
Implements CRUD operations for article

### Description of Task(s) to be completed?

-A user should be able to create articles and read/access articles created by other authors.
-A user should be able to delete and update articles that he/she has authored.
-An article should at minimum have a title, body, description and support for images.
-Given two different users create articles of the same title, the app should generate unique slugs such that the articles don't create conflict.
-Article slugs should be automatically added to articles if they do not exist.
-Timestamped articles

### What are the relevant PT stories?
[#165305266](https://www.pivotaltracker.com/story/show/165305266)

### How should this be manually tested?

- [x] **Clone this repository on local machine and CD into project folder**

`$ git clone git@github.com:andela/ah-the-jedi-backend.git`

`$ cd ah-the-jedi-backend`

- [x] **Check out to the PR's branch**

`$ git checkout ft-create-article-165305266`

- [x] **Create and activate virtual environment**

`$ python3 -m venv venv`

`$ source .venv/bin/activate`

- [x] **Install app dependencies from requirements.txt**

`$ pip install -r api/requirements.txt`

- [x] **Make migrations and run the server**

`$ python manage.py makemigrations`

`$ python manage.py migrate`

`$ python manage.py runserver`

**Use postman to test the endpoints**

- [x] **Sign up a user**

`$ git clone https://github.com/andela/ah-the-jedi-backend.git`

- [x] **Test the endpoints**

`POST` Create article endpoint `<you-local-host>/api/articles/`

*Screenshot*

Screenshot 2019-05-01 at 13 30 23

`GET` Get all articles endpoint `<you-local-host>/api/articles/`

*Screenshot*

Screenshot 2019-05-01 at 13 30 49

`GET` Get one article endpoint `<you-local-host>/api/articles/{article-slug}/`

*Screenshot*

Screenshot 2019-05-01 at 13 31 42

`DELETE` Delete article endpoint `<you-local-host>/api/articles/{article-slug}/`

*Screenshot*

Screenshot 2019-05-01 at 13 32 57

`PUT` Update article endpoint `<you-local-host>/api/articles/{article-slug}/`

*Screenshot*

Screenshot 2019-05-01 at 13 32 05

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -5,4 +5,5 @@ omit =
     config/*
     test/*
     manage.py
+    authors/apps/articles/utils.py
```
Comment (katherine95) [inline .coveragerc]: Add a line at the end of file.

**Review by AlvinMugambi — COMMENTED** [review]
You can look to add the environment variables to the .env-example for easier set up

File: authors/apps/articles/views.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        """
+        Function to handle permissions
+        """
+        try:
+            return [permission() for permission in self.permission_classes_by_action[self.action]]
+        except KeyError:
+            return [permission() for permission in self.permission_classes]
+
+
+    def create(self, request):
```
Comment (AlvinMugambi) [inline authors/apps/articles/views.py:49]: You can look to validate for duplicate articles where a user should not be able to add the same article twice.

**Review by jkamz — DISMISSED** [review]
Very good work on this task @Elisha-Misoi. However, please ensure that you include a 'following' field in the JSON object returned by an article. 
Additionally, please ensure that the slug is changed automatically when the title is updated.

File: authors/apps/articles/views.py (hunk 2 above)
Comment (ElishaMisoi) [reply authors/apps/articles/views.py:49]: Done

**Comment by ElishaMisoi** [comment]
> Very good work on this task @Elisha-Misoi. However, please ensure that you include a 'following' field in the JSON object returned by an article.
> Additionally, please ensure that the slug is changed automatically when the title is updated.

The change is in the code, if the User object has 'following' or 'bio' it will automatically be returned in the json response

**Review by Benkimeric — APPROVED** [review]
LGTM

**Review by jkamz — APPROVED** [review]

**Review by Siffersari — APPROVED** [review]

**Review by AlvinMugambi — APPROVED** [review]

**Review by katherine95 — APPROVED** [review]
LGTM.

**Review by rnjane — APPROVED** [review]

**Pull request merged**